### PR TITLE
Fix delete and add error message

### LINKED ISSUE #256: [PE-D][Tester B] Wrong command output when using `DELETE` with start index as `0`
**Description:**

When the `DELETE` command is performed with `0` as the start index range, the following output is shown:

```
Invalid command format!
Invalid range: The start index must be less than or equal to the end index.
```

However, this is not the correct error message as the start index 0 _is_ less than the end index.

**Steps to reproduce:**

```
add n/John Doe e/a@example.com g/M a/30
add n/John Doe e/b@example.com g/M a/30
delete 0-2
```

**Expected behavior:**

The correct error message would be something like:

```
Please enter a start index within the size of the list. (Your list has 2 entries.)
```

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: MiloTruck/ped#2

### LINKED ISSUE #261: [PE-D][Tester E] Add command: Name Parameter constraints not specified
### Description

The remarks for add command in User Guide does not specify that it cannot take in special characters but when typed in command prompt, it throws an error.

### Suggested Improvement

Mention in User Guide that name parameter cannot take in special characters.

-------------
Labels: `type.DocumentationBug` `severity.High`
original: ritvi12/ped#5

### LINKED ISSUE #269: [PE-D][Tester E] Delete Command: Confusing Format
### Description

The delete command specifies that when giving range it should be `INDEX-INDEX`. This can be confusing to user as they do not know when it should bounds of the range and in which order, lower bound - upper bound or the other way round. Unless they read remarks or examples.

### Suggested Improvements

Make the `INDEX-INDEX` `LowerBound-UpperBound` to avoid any such confusion.

-------------
Labels: `type.DocumentationBug` `severity.High`
original: ritvi12/ped#7

### LINKED ISSUE #273: [PE-D][Tester E] User Guide: Inconsistent Notes about Command Format
### Description

The notes about Command Format in User Guide mentions that the parameters in square brackets are optional. However, blank delete command throws an error, despite the command format having square brackets around the INDEX parameters.

### Suggested Improvement

Remove square brackets in the delete command format or edit the remark to specify the special case of delete command

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: ritvi12/ped#8

### LINKED ISSUE #275: [PE-D][Tester F] Alphanumeric names not respected
As per title

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: yhanyi/ped#11

### LINKED ISSUE #281: [PE-D][Tester E] Delete Command: Multiple Indices UG Remark
### Description

In case of multiple indices given, even if one of the indices are out of bounds of the list, then it throws the error of invalid index. 

But no such mention of it in the user guide.

### Suggested Improvement

Can add a remark on this behaviour for removing ambiguity

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: ritvi12/ped#10

### PR DESCRIPTION
**Pull request by wujiaweijoanna**
Fixes #281 
Fixes #275 
Fixes #273 
Fixes #269 
Fixes #261
Fixes #256

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -145,10 +145,11 @@ Adds a person to the address book.
...
 Format: `add n/NAME e/EMAIL g/GENDER a/AGE [d/DETAIL] [t/STUDY_GROUP_TAG]…​`
 
-* Email, gender and study group tags are **case-insensitive**
-* Valid entries for gender are M/m/F/f
-* Age must be a positive integer less than 150
-* Two contacts are considered **duplicates** if they share the same **email** address
+* Email, gender and study group tags are **case-insensitive**.
+* Valid entries for names must consist only of letters; numbers and special characters are not allowed. Each part of the name should be separated by exactly one space, with no consecutive spaces permitted.
+* Valid entries for gender are M/m/F/f.
+* Age must be a positive integer less than 150.
```
Comment (k-hian) [inline docs/UserGuide.md:151]: I believe @wkxcass 's pr is making some changes here. The wording is different and probably more specific/correct. Do you want to take a look at #313 and compare?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -188,16 +189,16 @@ Examples:
 
 Deletes the specified persons from the address book.
 
-Format: `delete [INDEX | INDEX-INDEX]…`
+Format: `delete INDEX | LowerBound-UpperBound [INDEX]… [LowerBound-UpperBound]…`
```
Comment (k-hian) [inline docs/UserGuide.md:192]: Is making one compulsory field necessary? It makes the command format look long and cluttered.

Additionally, `edit` command has a similar situation, which was solved by adding a remark stating that at least one optional field is required.

File: src/main/java/seedu/address/logic/parser/DeleteCommandParser.java [hunk 3]
```diff
@@ -22,6 +22,9 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public DeleteCommand parse(String args) throws ParseException {
+        if (args.trim().isEmpty()) {
```
Comment (k-hian) [inline src/main/java/seedu/address/logic/parser/DeleteCommandParser.java:25]: Are you changing the behaviour of the delete command here? There are some changes that are disallowed by feature freeze.

File: src/main/java/seedu/address/model/person/Name.java [hunk 4]
```diff
@@ -9,8 +9,9 @@
  */
 public class Name {
 
-    public static final String MESSAGE_CONSTRAINTS = "Names should only contain alphanumeric characters and spaces, "
-            + "and it should include given name and surname!";
+    public static final String MESSAGE_CONSTRAINTS = "Name must consist only of letters.\n"
+            + "Each name part is separated by exactly one space.\n"
```
Comment (k-hian) [inline src/main/java/seedu/address/model/person/Name.java:13]: Would `Name should only contain alphabet characters and spaces` be a better description? Also not very sure if adding the `Each name part is separated by exactly one space` is allowed

**Review by k-hian — COMMENTED** [review]
Some concerns regarding the changes. Perhaps you can take a look.

File: docs/UserGuide.md (hunk 1 above)
Comment (wkxcass) [reply docs/UserGuide.md:151]: I think best case is I merge my UG pr first, then @wujiaweijoanna can merge any related conflicts afterwards! (Need reviews first hahaha)

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 5]
```diff
@@ -21,7 +21,7 @@ public class DeleteCommand extends Command {
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX_1\n"
+            + "Parameters: INDEX_1 | LOWER_BOUND-UPPER_BOUND\n"
```
Comment (k-hian) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:24]: Is there a need for so many parameters here? 

Shouldn't it just be
```
[INDEX | LOWER-UPPER] ...
```

**Review by k-hian — COMMENTED** [review]
Additional nits.

File: docs/UserGuide.md (hunk 1 above)
Comment (wkxcass) [reply docs/UserGuide.md:151]: Update: merged my UG pr!

**Review by k-hian — COMMENTED** [review]
LGTM

**Comment by wkxcass** [comment]
LGTM!

**Pull request merged**